client/landing/gutenboarding/onboarding-block/vertical-select/index.tsx

```diff
@@ -61,13 +61,14 @@ const VerticalSelect: FunctionComponent< InjectedStepProps > = ( {
 	);
 
 	const { siteVertical } = useSelect( select => select( STORE_KEY ).getState() );
-	const { setSiteVertical } = useDispatch( STORE_KEY );
+	const { setSiteVertical, resetSiteVertical } = useDispatch( STORE_KEY );
 
-	const showSuggestions = () => setsuggestionsVisibility( true );
-	const hideSuggestions = () => setsuggestionsVisibility( false );
-
-	const handleSuggestionChangeEvent = ( e: React.ChangeEvent< HTMLInputElement > ) =>
+	const handleSuggestionChangeEvent = ( e: React.ChangeEvent< HTMLInputElement > ) => {
 		setInputValue( e.target.value );
+		if ( e.target.value !== inputValue && ! dirty ) {
```

Comment:
Should this whole conditional maybe go before `setInputValue( e.target.value );`, so people aren't lead to believe that `setInputValue( e.target.value )` sets `inputValue = e.target.value` immediately, thus causing `e.target.value !== inputValue` to always evaluate to `true`? :thinking: